transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoHandler.java

```diff
@@ -268,9 +296,7 @@ public void destroy() {
         long udata = UserData.encode(RINGFD_ID, Native.IORING_OP_NOP, (short) 0);
         submissionQueue.addNop((byte) Native.IOSQE_IO_DRAIN, udata);
         // ... but only wait for 200 milliseconds on this
-        udata = UserData.encode(RINGFD_ID, Native.IORING_OP_LINK_TIMEOUT, (short) 0);
-        submissionQueue.addLinkTimeout(TimeUnit.MILLISECONDS.toNanos(200), udata);
-        submissionQueue.submitAndWait();
+        submitAndWaitWithTimeout(submissionQueue, true, TimeUnit.MILLISECONDS.toNanos(200));
```

Comment:
I don't think linking the SQE works like you expect here. The [man page](https://www.mankier.com/2/io_uring_enter#Description-IOSQE_IO_LINK) states:

> When this flag is specified, the SQE forms a link with the next SQE in the submission ring. That next SQE will not be started before the previous request completes.

However, the timeout comes after the nop. For the TIMEOUT to link to the NOP, I think the timeout has to come first.

Reply:
So you are saying the code before also did not work as expected correct ? As I only moved it

Reply:
(which seems like the case from the manpage)

Reply:
Yeah, looks like it.

Reply:
Actually the order was correct as we are talking about `IORING_OP_LINK_TIMEOUT` here.. That said there was still a bug as we did miss to add `IOSQE_LINK` to the `IORING_OP_NOP` above. I opened another PR to fix this in isolation: https://github.com/netty/netty/pull/14717